samples/Microsoft.Bot.Samples.Ai.Luis.Translator/Startup.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+            Configuration = builder.Build();
+        }
+
+        public IConfigurationRoot Configuration { get; }
+
+        private void SetLanguage(ITurnContext context, string language) => context.GetConversationState<CurrentUserState>().Language = language;
+        private void SetLocale(ITurnContext context, string locale) => context.GetConversationState<CurrentUserState>().Locale = locale;
+
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddBot<LuisTranslatorBot>(options =>
+            {
+            options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);
+
+            string luisModelId = "<Your Model Here>";
+            string luisSubscriptionKey = "<Your Key here>";
+            Uri luisUri = new Uri("https://westus.api.cognitive.microsoft.com/luis/v2.0/apps/");
+            var luisModel = new LuisModel(luisModelId, luisSubscriptionKey, luisUri); 
+
+            // If you want to get all intents scorings, add verbose in luisOptions
+            var luisOptions = new LuisRequest { Verbose = true };
+            Dictionary<string, List<string>> patterns = new Dictionary<string, List<string>>();
+            patterns.Add("fr", new List<string> { "mon nom est (.+)" });//single pattern for fr language
+            var middleware = options.Middleware;
+            middleware.Add(new ConversationState<CurrentUserState>(new MemoryStorage()));
+            middleware.Add(new TranslationMiddleware(new string[] { "en" }, "<your translator key here>", patterns, GetActiveLanguage, SetActiveLanguage));
+           middleware.Add(new LocaleConverterMiddleware(GetActiveLocale, SetActiveLocale, "en-us", new LocaleConverter()));
+            middleware.Add(new LuisRecognizerMiddleware(luisModel, luisOptions: luisOptions));
+            });
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseDefaultFiles()
+                .UseStaticFiles()
+                .UseBotFramework();
+        }
+
+        protected bool IsSupportedLanguage(string language) => _supportedLanguages.Contains(language);
+        protected async Task<bool> SetActiveLanguage(ITurnContext context)
```

Comment:
It seems very weird to have these methods in startup.cs. 

Consider moving them into a dedicated static class to make orginization more obvious.

Reply:
Moved to another class